app/src/main/java/org/sopt/sample/presentation/login/signup/SignUpViewModel.kt

```diff
@@ -5,12 +5,19 @@ import kotlinx.coroutines.launch
 import org.sopt.sample.data.auth.model.SignUpRequest
 import org.sopt.sample.data.auth.repository.AuthRepository
 import org.sopt.sample.util.addSourceList
+import java.util.regex.Pattern
 
 
 class SignUpViewModel(private val authRepository: AuthRepository) : ViewModel() {
+    //id - 6~10글자 영문 숫자 포함
+    //pw - 6~12글자 영문,숫자,특수문자 포함
+    private val idRegex = "^(?=.*[a-zA-Z]+)(?=.*[0-9]+).{6,10}$"
+    private val pwRegex = "^(?=.*[a-zA-Z]+)(?=.*[0-9]+)(?=.*[!@#$%^&*()~`<>?:']+).{6,12}$"
+    private val idMatcher: Pattern = Pattern.compile(idRegex)
+    private val pwMatcher: Pattern = Pattern.compile(pwRegex)
```

Comment:
정규표현식 compile 비용이 생각보다 세서 한번만 만드는게 좋을 것 같아

```suggestion
    companion object {
        private val REGEX_ID_PATTERN = Pattern.compile("^(?=.*[a-zA-Z]+)(?=.*[0-9]+).{6,10}$")
        private val REGEX_PASSWORD_PATTERN = Pattern.compile("^(?=.*[a-zA-Z]+)(?=.*[0-9]+)(?=.*[!@#$%^&*()~`<>?:']+).{6,12}$")
    }
```